### PR TITLE
New engine: Napalm syslog

### PR DESCRIPTION
**Pull request by mirceaulinic**
### What does this PR do?

Defines a new engine taking events from a third party library called napalm-logs and publishes them on the Salt bus. The events format respects the OpenConfig and IETF standards.

### CODE REVIEW

File: salt/engines/napalm_syslog.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+'''
+from __future__ import absolute_import
+
+# Import python stdlib
+import json
+import logging
+
+# Import third party libraries
+import zmq
```
Comment (cachedout) [inline salt/engines/napalm_syslog.py]: Not everybody uses Salt with ZeroMQ. This should be gated.

**Review by cachedout — CHANGES_REQUESTED** [review]
I really like this!

Just one small change requested though. See the in-line comment. This should be really nice to get in. Strong work!

File: salt/engines/napalm_syslog.py (hunk 1 above)
Comment (mirceaulinic) [inline salt/engines/napalm_syslog.py]: No problem - added in https://github.com/saltstack/salt/pull/40458/commits/75c64bfe453d86f2f19a41fac5d2ccc49a40739b.

**Comment by mirceaulinic** [comment]
Hi @cachedout - I have solved the changes requested.

**Review by cachedout — APPROVED** [review]

**Pull request merged**